homeassistant/components/blueprint/websocket_api.py

```diff
@@ -84,3 +101,81 @@ async def ws_import_blueprint(hass, connection, msg):
             },
         },
     )
+
+
+@websocket_api.async_response
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "blueprint/save",
+        vol.Required("domain"): cv.string,
+        vol.Required("path"): cv.path,
+        vol.Required("data"): cv.string,
+        vol.Optional("source_url"): cv.url,
+    }
+)
+async def ws_save_blueprint(hass, connection, msg):
+    """Save a blueprint."""
+
+    path = msg["path"]
+    domain = msg["domain"]
+
+    def create_file():
+        """Create blueprint file."""
+        blueprint_path = pathlib.Path(
+            hass.config.path(BLUEPRINT_FOLDER, domain, f"{path}.yaml")
+        )
+        if blueprint_path.exists():
+            raise HomeAssistantError("File already exists")
+
+        blueprint = models.Blueprint(yaml.parse_yaml(msg["data"]))
+        if msg.get("source_url"):
```

Comment:
`get` is a little weird, because we then also test the returned value for truthiness.

```suggestion
        if "source_url" in msg:
```